### ISSUE TITLE
Stack overflow error occurs while processing  class Collection1 : IEnumerable<KeyValuePair<string, object>>

### ISSUE THREAD

**Issue by alexeybut** [microsoft/node-api-dotnet#253]
The error:
```
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error : Stack overflow. [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.SymbolDisplay.AbstractSymbolDisplayVisitor.Initialize(Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder`1<Microsoft.CodeAnalysis.SymbolDisplayPart>, Microsoft.CodeAnalysis.SymbolDisplayFormat, Boolean, Microsoft.CodeAnalysis.SemanticModel, Int32, Boolean) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.SymbolDisplayVisitor.Initialize(Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder`1<Microsoft.CodeAnalysis.SymbolDisplayPart>, Microsoft.CodeAnalysis.SymbolDisplayFormat, Boolean, Microsoft.CodeAnalysis.SemanticModel, Int32, Boolean) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.SymbolDisplayVisitor.GetInstance(Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder`1<Microsoft.CodeAnalysis.SymbolDisplayPart>, Microsoft.CodeAnalysis.SymbolDisplayFormat, Microsoft.CodeAnalysis.SemanticModel, Int32) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.SymbolDisplay.PopulateDisplayParts(Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder`1<Microsoft.CodeAnalysis.SymbolDisplayPart>, Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.SemanticModel, Int32, Microsoft.CodeAnalysis.SymbolDisplayFormat, Boolean) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.SymbolDisplay.ToDisplayString(Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.SemanticModel, Int32, Microsoft.CodeAnalysis.SymbolDisplayFormat, Boolean) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.Symbol.ToString() [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.GetTypeSymbolFullName(Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
```
Code to reproduce:
```
    [JSExport]
    public class Class2
    {
        public Collection1 Items { get; set; }
    }

    public class Item
    {
    }

    public class Collection1 : IEnumerable<KeyValuePair<string, object>>
    {
        public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
        {
            throw new NotImplementedException();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            throw new NotImplementedException();
        }
    }
```
Sample project is attached:
[ConsoleApp1.zip](https://github.com/microsoft/node-api-dotnet/files/14823035/ConsoleApp1.zip)
Version 0.6.3
AOT is on.
Command:
`
dotnet publish -c DEBUG -r win-x64
`
Seems is linked with #249.

**Comment by alexeybut** [comment]
@jasongin Hi, I have tried the fixed version from the master but seems the issue is still reproduced. I tried the same sample console app from the attachment.

**Comment by jasongin** [comment]
The C# build server can cache the source-generator, and the cache does not get invalidated by package version updates. So make sure you have shutdown the build server. [The doc](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-server?view=vs-2022#shut-down-or-disable-msbuild-server) suggests `dotnet build-server shutdown` but I have found even that does not always work if VS is keeping it alive. An alternative is to simply reboot.

You can check the verbose MSBuild log to confirm what version of the generator it is actually running.

**Comment by alexeybut** [comment]
@jasongin I have already used dotnet build-server shutdown command.
I restarted the system, cleared the nuget packages from the cache. I have tried the version 0.6.5 from nuget. Also I have built 0.7.0 version from the latest main and tried it. 
Anyway I get:
```
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error : Stack overflow. [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at System.Reflection.Internal.MemoryBlock.BinarySearchReference(Int32, Int32, Int32, UInt32, Boolean) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at System.Reflection.Internal.MemoryBlock.BinarySearchReferenceRange(Int32, Int32, Int32, UInt32, Boolean, Int32 ByRef, Int32 ByRef) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at System.Reflection.Metadata.Ecma335.CustomAttributeTableReader.GetAttributeRange(System.Reflection.Metadata.EntityHandle, Int32 ByRef, Int32 ByRef) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at System.Reflection.Metadata.MetadataReader.GetCustomAttributes(System.Reflection.Metadata.EntityHandle) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.PEModule.FindTargetAttribute(System.Reflection.Metadata.MetadataReader, System.Reflection.Metadata.EntityHandle, Microsoft.CodeAnalysis.AttributeDescription, Boolean ByRef) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.PEModule.HasNullableAttribute(System.Reflection.Metadata.EntityHandle, Byte ByRef, System.Collections.Immutable.ImmutableArray`1<Byte> ByRef) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PETypeParameterSymbol.GetNullableAttributeValue() [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PETypeParameterSymbol.get_HasNotNullConstraint() [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PETypeParameterSymbol.get_IsNotNullable() [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations+LazySubstitutedType.<GetResolvedAnnotation>g__getResolvedAnnotationCore|11_0() [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations+LazySubstitutedType.GetResolvedAnnotation(Microsoft.CodeAnalysis.CSharp.NullableAnnotation) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.NullableAnnotationExtensions.ToPublicAnnotation(Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.NullableAnnotationExtensions+<>c.<GetPublicSymbols>b__13_0(Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at System.Collections.Immutable.ImmutableArray.CreateRange[[Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations, Microsoft.CodeAnalysis.CSharp, Version=4.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35],[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations>, System.Func`2<Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations,System.__Canon>) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.ImmutableArrayExtensions.SelectAsArray[[Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations, Microsoft.CodeAnalysis.CSharp, Version=4.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35],[System.__Canon, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations>, System.Func`2<Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations,System.__Canon>) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.NamedTypeSymbol.Microsoft.CodeAnalysis.INamedTypeSymbol.get_TypeArguments() [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.BuildBaseTypeAndInterfaces(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
C:\Program Files\dotnet\sdk\8.0.202\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Microsoft.JavaScript.NodeApi.Generator.SymbolExtensions.<BuildBaseTypeAndInterfaces>g__BuildType|15_0(Microsoft.CodeAnalysis.INamedTypeSymbol, Microsoft.CodeAnalysis.INamedTypeSymbol) [X:\NodeJS\ConsoleApp1\ConsoleApp1.csproj]
......
```
I use the same sample project from this issue.

**Comment by jasongin** [comment]
Sorry, I can indeed still reproduce the stack overflow with your code.

I had added a test case that reproduced what I thought was the same problem with simplified code, but apparently it's slightly different.

**Comment by alexeybut** [comment]
@jasongin do you use the version 0.6.5 from nuget.org?

**Comment by alexeybut** [comment]
[ConsoleApp2.zip](https://github.com/microsoft/node-api-dotnet/files/14910319/ConsoleApp2.zip)
Please unzip the archive and run publish.cmd to reproduce the issue.
The project is the same I attached before, but the version 0.6.6 is referenced.

**Comment by jasongin** [comment]
My last comment was confirming that I can reproduce the error with the latest version.
#266 will fix it.

**Comment by alexeybut** [comment]
@jasongin Sorry, I looked through it ;) Thanks.

**Comment by jasongin** [comment]
Fixed in `0.6.8`, which should be published in an hour or so.

And I apologize: I should have tested against the code you originally reported before doubting you.
